### PR TITLE
Tigris example with Next.js

### PR DESCRIPTION
**Pull request by adilansari**
## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: examples/with-tigris/.env.example [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Enter your tigris uri, ex :- localhost:8081, api.preview.tigrisdata.cloud etc.
+TIGRIS_URI=
+
+# Client credentials, if using auth, can be generated from Tigris cloud console.
+# See: https://docs.tigrisdata.com/auth
+TIGRIS_CLIENT_ID=
+TIGRIS_CLIENT_SECRET=
```
Comment (balazsorban44) [inline examples/with-tigris/.env.example:7]: Given this file has a secret, let's rename to `.env.local.example`

**Comment by Man-from-earth25** [comment]
**WellDone Bro**

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
The `pnpm-lock.yaml` file should be untouched by examples, could you revert those changes?

File: examples/with-tigris/.gitignore [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+.dynamodb/
+
+# TernJS port file
+.tern-port
+
+# IDE
+.idea/
+
+# next env
+next-env.d.ts
```
Comment (balazsorban44) [inline examples/with-tigris/.gitignore:109]:
```suggestion
# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.

# dependencies
/node_modules
/.pnp
.pnp.js

# testing
/coverage

# next.js
/.next/
/out/

# production
/build

# misc
.DS_Store
*.pem

# debug
npm-debug.log*
yarn-debug.log*
yarn-error.log*
.pnpm-debug.log*

# local env files
.env*.local

# vercel
.vercel

# typescript
*.tsbuildinfo
next-env.d.ts
```

Let's follow other examples

File: examples/with-tigris/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (balazsorban44) [inline examples/with-tigris/LICENSE:1]: This file should be removed

File: examples/with-tigris/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "@tigrisdata/core": "^1.0.0-beta.16",
+    "next": "latest",
+    "react": "18.2.0",
+    "react-dom": "18.2.0"
+  },
+  "devDependencies": {
+    "@types/node": "18.11.2",
+    "@types/react": "18.0.21",
+    "@types/react-dom": "18.0.6",
+    "@typescript-eslint/eslint-plugin": "^5.40.0",
```
Comment (balazsorban44) [inline examples/with-tigris/package.json:22]:
```suggestion
```

File: examples/with-tigris/package.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  "private": true,
+  "scripts": {
+    "predev": "APP_ENV=development npm run setup",
+    "dev": "next dev",
+    "build": "next build",
+    "postbuild": "APP_ENV=production npm run setup",
+    "start": "next start",
+    "clean": "rm -rf node_modules .next",
+    "setup": "npx ts-node scripts/setup.ts"
```
Comment (balazsorban44) [inline examples/with-tigris/package.json:10]: These have assumptions that eg `ts-node` is installed and I believe the way APP_ENV is used it won't work cross OS. can we rather add a comment about running `scripts/setup.ts` in the readme?
```suggestion
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
```

File: examples/with-tigris/pages/_document.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { Html, Head, Main, NextScript } from 'next/document'
```
Comment (balazsorban44) [inline examples/with-tigris/pages/_document.tsx:1]: This file should be removed, seems unnecessary for the sake of the example

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, added some comments. Could you address them?

File: examples/with-tigris/pages/_document.tsx (hunk 6 above)
Comment (adilansari) [reply examples/with-tigris/pages/_document.tsx]: removed

File: examples/with-tigris/package.json (hunk 5 above)
Comment (adilansari) [reply examples/with-tigris/package.json]: Added a note in readme regarding custom `dev` and `build` scripts. This is demonstrating a feature that lets user bootstrap schema automatically, they can opt out of it. :)

File: examples/with-tigris/package.json (hunk 4 above)
Comment (adilansari) [reply examples/with-tigris/package.json]: removed

File: examples/with-tigris/LICENSE (hunk 3 above)
Comment (adilansari) [reply examples/with-tigris/LICENSE]: removed

**Comment by adilansari** [comment]
> Thanks, added some comments. Could you address them?

Addressed! Ready for another round of review. Thanks for the help.

File: examples/with-tigris/pages/_app.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import '../styles/globals.css'
+import type { AppProps } from 'next/app'
+import React from 'react'
```
Comment (balazsorban44) [inline examples/with-tigris/pages/_app.tsx:3]:
```suggestion
```

**Review by balazsorban44 — APPROVED** [review]

**Comment by adilansari** [comment]
@balazsorban44 thanks for approving. How do I get it merged?

**Comment by balazsorban44** [comment]
Merged manually. :+1: Usually PRs are merged when everything passes, but this PR was opened from an organization fork, which GitHub cannot currently auto-merge because of a bug. https://github.com/orgs/community/discussions/5634

**Pull request merged**